sync3/handler/handler.go

```diff
@@ -802,6 +802,15 @@ func (h *SyncLiveHandler) OnExpiredToken(p *pubsub.V2ExpiredToken) {
 	h.ConnMap.CloseConnsForDevice(p.UserID, p.DeviceID)
 }
 
+func (h *SyncLiveHandler) OnInvalidateRoom(p *pubsub.V2InvalidateRoom) {
+	ctx, task := internal.StartTask(context.Background(), "OnInvalidateRoom")
+	defer task.End()
+
+	logger.Warn().Msgf("DMR: invalidate room %s", p.RoomID)
+	h.GlobalCache.OnInvalidateRoom(ctx, p.RoomID)
+	h.Dispatcher.OnInvalidateRoom(ctx, p.RoomID)
```

Comment:
The dispatcher usually calls the global cache e.g:
```go
	// global listeners (invoke before per-user listeners so caches can update)
	listener := d.userToReceiver[DispatcherAllUsers]
	if listener != nil {
		listener.OnEphemeralEvent(ctx, roomID, ephEvent)
	}
```

Reply:
6f37ac0